### PR TITLE
Unit test max block size reached

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added unit test for max block size reached edge case

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: process/block/preprocess/transactions_test.go [hunk 1]
```diff
@@ -51,6 +51,15 @@ func shardedDataCacherNotifier() dataRetriever.ShardedDataCacherNotifier {
 					if reflect.DeepEqual(key, []byte("tx2_hash")) {
 						return &transaction.Transaction{Nonce: 10}, true
 					}
+					if reflect.DeepEqual(key, []byte("tx_hash1")) {
```
Comment (iulianpascalau) [inline process/block/preprocess/transactions_test.go:54]: This could have worked better with:
```
if bytes.Equal(key, []byte("tx_hash1")) {
```

can remain as it is

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
